src/UI/TreeView.re

```diff
@@ -177,6 +235,22 @@ module Make = (Model: TreeModel) => {
     let scrollTop =
       scrollTop |> Utility.clamp(~lo=0, ~hi=itemHeight * count - menuHeight);
 
+    let%hook () =
+      Hooks.effect(
+        If((!=), focus),
+        () => {
+          focus
+          |> Option.bind(nodeOffsetByPath(tree))
+          |> Option.iter(offset => {
+               // center focused item
+               setScrollTop(_ =>
+                 offset * itemHeight - menuHeight / 2
+               )
+             });
+          None;
+        },
+      );
```

Comment:
This doesn't really do its job, because `focus` changes structurally whenever part of the tree is loaded, because we don't reuse ids and of course because the tree actually changes structurally.

Structural comparison on `FsTreeNode`'s is also amazingly inefficient (exponential?). If we preserve node ids we can just compare those, but I don't think the effect hook approach is viable here anyway since we need to distinguish the origin of the change (`BufferEnter` vs anything else), which we can't do here.

Reply:
Fixed by allowing scrollTop to be controlled externally